src/main/java/main/Duke.java

```diff
@@ -0,0 +1,189 @@
+import tasks.Event;
+
+
+public class Duke {
+
+    // Function to mark or unmark tasks
+    public static void userMarkOrUnmark(String command, String line, ArrayList<Task> list, int index, Task t) {
+
+        // User enters Mark, proceed to check if index is valid. If valid, then mark the task number
+        if (command.equals("mark")) {
+            try {
+                index = Integer.parseInt(line.substring(5));
+            } catch (NumberFormatException e) {
+                System.out.println("Task number is not a valid number");
+                return;
+            }
+            t = list.get(index - 1);
+            t.isDone =  true;
+            System.out.println("Nice! I've marked this task as done:");
+        }
+
+        // User enters unmark, proceed to check if index is valid. If valid, then unmark the task number
+        else {
+            try {
+                index = Integer.parseInt(line.substring(7));
+            } catch (NumberFormatException e) {
+                System.out.println("Task number is not a valid number");
+                return;
+            }
+            t = list.get(index - 1);
+            t.isDone =  false;
+            System.out.println("OK, I've marked this task as not done yet:");
+        }
+
+        System.out.println(t);
+        System.out.println("____________________________________________________________");
+    }
+
+    // Function to print the list of tasks
+    public static void userList(String line, ArrayList<Task> list) {
+            for (Task task : list) {
+                if (task == null) {
+                    break;
+                }
+                System.out.println(task.index + ". " + task);
+            }
+            System.out.println("____________________________________________________________");
+    }
+
+    // Function to say bye
+    public static void userBye() {
+        System.out.println("Bye human. Come back soon !");
+    }
+
+    // Function to print an unknown command from user
+    public static void userWrongCommand() {
+        System.out.println("No suitable command found. Please try again!");
+    }
+
+    // Function to check the minimum arguments supplied for each task
+    public static boolean checkMinimumArguments(String[] splitLine, int number) {
+        try {
+            if (splitLine.length < number) {
+                throw new DukeException("Invalid Syntax! Please try again!");
+            }
+        } catch (DukeException e) {
+            System.out.println("Minimally " + number + " arguments, please try again!");
+            return true;
+        }
+        return false;
+    }
+
+    // Start of user input
+    public static void userInput() throws DukeException {
+        Scanner scanner = new Scanner(System.in); // Declared a scanner object
+        String line; // Declared a string object to take in user input
+        ArrayList<Task> list = new ArrayList<>(); // Declared a dynamic list with Task data type
```

Comment:
Try not to repeat in comments information that is already obvious from the code. This will be covered in future lectures.